### DISCUSSION TITLE
Is Python support removed in the newer version of the Docker Image?

### DISCUSSION THREAD

**Discussion by gpanagiotidis** [argoproj/argo-cd#8939] [Q&A]
I upgraded to the new version of argocd and noticed that my plugin stopped working as there is no longer any python binary present in the filesystem. Is there a valid reason for such a big change?

**Comment by crenshaw-dev** [comment]
Can you specify the before and after versions? I'm not aware of any change that should have caused this. But I can confirm that Python doesn't appear to be on the 2.3.3 image.

**Comment by gpanagiotidis** [comment]
Sorry for the late reply. In v2.1 python was present in the image. This merge https://github.com/argoproj/argo-cd/pull/7947 not only removed python deps (by not installing python3-pip), but it removed python in general, as python was being installed as a dependency to python3-pip.

This is a very big change in my opinion. People could have been running custom python plugins and there is not even a mention in the changelog about it.

**Reply by crenshaw-dev** [reply]
Excellent find. We should definitely update release notes. I'd rather avoid re-introducing Python for image size and security reasons. But we should definitely be clear that the change happened.

Opened a PR: https://github.com/argoproj/argo-cd/pull/9090
